app/controllers/miq_policy_controller/miq_actions.rb

```diff
@@ -138,11 +138,11 @@ def action_field_changed
   def action_tag_pressed
     @edit = session[:edit]
     @action = @edit[:action_id] ? MiqAction.find_by_id(@edit[:action_id]) : MiqAction.new
-    tag_name = params[:id].split('__')[1]
+    id = TreeBuilder.from_cid(params[:id].split('-')[1])
```

Comment:
`from_cid` should be available in controllers without the `TreeBuilder.`..

Or even better, @ZitaNemeckova don't we have a TreeBuilder method that gives you that parsed out id?

.. I'm thinking `_ id, _ = TreeBuilder.extract_node_model_and_id(params[:id])` (**if** it works with Classifications..)